sklearn/decomposition/kernel_pca.py

```diff
@@ -220,11 +220,31 @@ def _fit_transform(self, K):
         self.lambdas_ = self.lambdas_[indices]
         self.alphas_ = self.alphas_[:, indices]
 
-        # remove eigenvectors with a zero eigenvalue
+        # remove eigenvectors with a zero eigenvalue (null space) if required
         if self.remove_zero_eig or self.n_components is None:
             self.alphas_ = self.alphas_[:, self.lambdas_ > 0]
             self.lambdas_ = self.lambdas_[self.lambdas_ > 0]
 
+        # Maintenance note on Eigenvectors normalization
+        # ----------------------------------------------
+        # there is a link between
+        # the eigenvectors of K=Phi(X)'Phi(X) and the ones of Phi(X)Phi(X)'
+        # if v is an eigenvector of K
+        #                      then Phi(X)v  is an eigenvector of Phi(X)Phi(X)'
+        # if u is an eigenvector of Phi(X)Phi(X)'
+        #                      then Phi(X)'u is an eigenvector of Phi(X)Phi(X)'
+        #
+        # At this stage our self.alphas_ (the v) have norm 1, we need to scale
+        # them so that eigenvectors in kernel feature space (the u) have norm=1
+        # instead
+        #
+        # We COULD scale them here:
+        #       self.alphas_ = self.alphas_ / np.sqrt(self.lambdas_)
+        #
+        # But the original choice was to perform that LATER when needed, in
+        # fit() and in transform(). We keep it to preserve the meaning of
+        # self.alphas_ across sk-learn versions.
```

Comment:
I was referring to the initial choice of the author (Mathieu Blondel) here, not to mine. Is there another way to express this ? Basically what I wanted to say here is that if I had to write this file from scratch I would have done the scaling here because that's much more intuitive. But since it was already done otherwise in previous sklearn versions, I dont change it for backwards compatibility reasons, even if IMO it is harder to read.

Maybe

```python
        # We COULD scale them here:
        #       self.alphas_ = self.alphas_ / np.sqrt(self.lambdas_)
        #
        # But it is instead done in fit() and in transform() since older 
        # versions of sklearn. We keep that logic to preserve the meaning of
        # self.alphas_ across sk-learn versions.
```

Reply:
I understand, but someone reading the code doesn't need/want to worry about the incremental changes. Hence "the original choice" doesn't make sense when you're reading the code at a given time, nor does "older versions of sklearn".

Any notion of temporality in the comments should be removed since it only adds cognitive overload IMO.

Reply:
fair enough.